plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -1095,7 +1096,34 @@ public void testPredicateOnDollarPrefixedField()
 
         assertThat(query("SELECT root.\"$field\" FROM test." + tableName + " WHERE root.\"$field\" = 'foo'"))
                 .matches("SELECT varchar 'foo'")
-                .isFullyPushedDown(); // isFullyPushedDown is misleading here because the connector does not pushdown for dollar prefixed fields
+                .isNotFullyPushedDown(FilterNode.class);
+
+        assertUpdate("DROP TABLE test." + tableName);
+    }
+
+    @Test
+    public void testProjectionPushdownMixedWithUnsupportedFieldName()
```

Comment:
With a deeply nested field - do we read them till the parent column or the whole root column ? Can we also assert the plan here - for the table scan to identify which column was selected - `queryAssertion` provides a way to validate the plan also I guess.

Reply:
> do we read them till the parent column or the whole root column ?

We read till parent column.

Eg:

```
Schema: root1 ROW(id INT, root2 ROW(id INT, root3 ROW(id INT, \"dotted.field\" INT, name VARCHAR)))

-- when projected root1.root2.root3.\"dotted.field\", then it will read root1.root2.root3 from database

```

Reply:
Added `TestMongoProjectionPushdownPlans`